.github/workflows/deps.yaml

```diff
@@ -167,7 +179,121 @@ jobs:
         with:
           name: linux_arm64_substrait
           path: deps/linux_arm64
+      - uses: actions/download-artifact@v4
+        with:
+          name: iceberg-v1.1.3-extension-linux_arm64
+          path: deps/linux_arm64
       - name: Push static libraries
         uses: stefanzweifel/git-auto-commit-action@v5
         with:
           commit_message: Re-build static libraries
+
+  iceberg:
+    name: Build extension binaries
+    uses: ./.github/workflows/iceberg_extension_distribution.yaml
+    with:
+      extension_name: iceberg
+      duckdb_version: v1.1.3
+      ci_tools_version: v1.1.3
+      exclude_archs: 'linux_amd64_gcc4'
+
+  test_darwin_amd64:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install avrocpp
+        shell: bash
+        run: |
+          brew install avro-cpp
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.23'
+
+      - name: Test loading the extensions
+        run: go test -run ^TestLoaded
+
+  test_darwin_arm64:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install avrocpp
+        shell: bash
+        run: |
+          brew install avro-cpp
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.23'
+
+      - name: Test loading the extensions
+        run: go test -run ^TestLoaded
+
+  test_linux_amd64:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install Avro C++ from Source
+        run: |
+          # Install dependencies
+          sudo apt-get update
+          sudo apt-get install -y libboost-all-dev libboost-iostreams-dev libboost-system-dev cmake
+
+          # Download Apache Avro C++ source
+          AVRO_VERSION="1.12.0"
+          wget https://archive.apache.org/dist/avro/avro-${AVRO_VERSION}/cpp/avro-cpp-${AVRO_VERSION}.tar.gz
+          tar -xzf avro-cpp-${AVRO_VERSION}.tar.gz
+          cd avro-cpp-${AVRO_VERSION}
+
+          # Build and install
+          sudo ./build.sh install
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.23'
+
+      - name: Test loading the extensions
+        run: go test -run ^TestLoaded
+
+  test_linux_arm64:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install Avro C++ from Source
```

Comment:
we should add `actions/cache` to make sure we don't have to repeat this step always

Reply:
where are these libraries being used anyways? we make them but i am not sure where we link them, why aren't we uplaoding static artifacts like the other `.a` files?

Reply:
we link them in cgo_static.go [here](https://github.com/sundeck-io/duckdb-go-extensions/pull/2/files/e31c1ba15256f2916710b13c6b4d265cfa96b982#diff-e2c7d6384647cf317be79d42f583210da3377427c003cb75e2593025658c842a)

Reply:
I tried to cache them but could not cache because of the below error. When debugged the paths were present. Further debugging may need admin access. 

```
      - name: Cache Avro Source
        uses: actions/cache@v3
        with:
          path: |
            /usr/local/lib
            /usr/local/include
          key: libs-${{ matrix.os }}-${{ env.AVRO_VERSION }}
          restore-keys: |
            libs-${{ matrix.os }}-${{ env.AVRO_VERSION }}
```

error:

<img width="1045" alt="Screenshot 2024-12-17 at 9 10 32 AM" src="https://github.com/user-attachments/assets/72998854-5404-45f9-b382-108b6a7f266c" />